ui/src/app/pages/workspace/create-billing-account-modal.tsx

```diff
@@ -1,76 +1,407 @@
+    return {
+      visibility: currentStep === 0 || currentStep === 4 ? 'hidden' : 'visible',
+      borderRadius: '50%',
+      height: '37px',
+      width: '37px',
+      opacity: currentStep === buttonStep ? '1.0' :  '0.2',
+      backgroundColor: currentStep === buttonStep ? colors.primary : '#333F52',
+      fontFamily: 'Montserrat',
+      textAlign: 'center',
+      lineHeight: '37px',
+      fontSize: '18px',
+      fontWeight: 600,
+      letterSpacing: '0',
+      color: '#FFFFFF',
+      marginRight: '10px',
+    };
+  }
+};
 
-        <ModalBody>
-          <FlexColumn style={{alignItems: 'flex-start'}}>
-            <img style={{width: '12rem', marginLeft: '-0.7rem'}} src='/assets/images/gcp_logo.png'/>
+export interface CreateBillingAccountState {
+  currentStep: number;
+  showBillingAccountDescription: boolean;
+  showSubmitInstruction: boolean;
+  userFullName: string;
+  userPhoneNumber: string;
+}
 
-            <TextColumn>
-              <div>Billing accounts are managed via Google Cloud Platform™ service.</div>
-              <div>Learn more on how to set up a billing account.</div>
-            </TextColumn>
+export interface Props {
+  onClose: Function;
+}
 
-            <Button type='primary'
-                    style={{fontWeight: 400, padding: '0 18px', height: '40px'}}
-                    onClick={() => window.open(supportUrls.createBillingAccount, '_blank')}>
-              Read Instructions
-            </Button>
-          </FlexColumn>
-        </ModalBody>
+const BillingConfirmItem = ({title, value, dataTestId}) => {
+  return <FlexRow id = {`${dataTestId}-wrapper`} style={{marginTop: '5px'}}>
+    <div style={{width: '170px'}}>{title}:</div>
+    <div data-test-id={dataTestId}>{value}</div>
+  </FlexRow>;
+};
 
-        <hr style={{width: '100%', backgroundColor: colors.primary, borderWidth: '0px', height: '1px',
-          marginTop: '1rem',
-          marginBottom: '0.5rem'}}/>
+export const CreateBillingAccountModal = ({onClose}: Props) => {
+  const {profile: {
+    contactEmail,
+    givenName,
+    familyName,
+    verifiedInstitutionalAffiliation,
+    username
+  }} = useStore(profileStore);
+  const [currentStep, setCurrentStep] = useState(0);
+  const [phoneNumber, setPhoneNumber] = useState();
+  const [invalidPhoneNumberInput, setInvalidPhoneNumberInput] = useState<boolean>(null);
+  const [useCreditCard, setUseCreditCard] = useState<boolean>(null);
+  const [nihFunded, setNihFunded] = useState<boolean>(null);
 
-        <ModalFooter style={{marginTop: 0, justifyContent: 'flex-start'}}>
-          <FlexColumn>
-            <TextColumn>
-              <p style={{marginTop: 0}}>Don't already have a Google Cloud account?</p>
-              <a href='https://cloud.google.com' target='_blank'>Create an account</a>
-            </TextColumn>
+  const validatePhoneNumber = (phoneInput: string) => {
+    const sanitizedPhone = phoneInput.replace(/\D/g, '');
```

Comment:
nit: kind of odd I can enter letters into the phone number and it considers it valid; stripping - ( and whitespace definitely good though

Reply:
yeah, I was thinking if user want to type 123-1234-(1234). So I just stripe everything and only kept numbers.